driver/pace/driver/initialization.py

```diff
@@ -39,7 +41,38 @@ def get_driver_state(
 
 
 @dataclasses.dataclass
-class BaroclinicConfig(InitializationConfig):
+class InitializationConfig:
+
+    type: str
+    config: Initializer
+    registry: ClassVar = Registry()
+
+    @classmethod
+    def register(cls, type_name):
+        return cls.registry.register(type_name)
+
+    @property
+    def start_time(self) -> datetime:
+        return self.config.start_time
+
+    def get_driver_state(
+        self,
+        quantity_factory: pace.util.QuantityFactory,
+        communicator: pace.util.CubedSphereCommunicator,
+    ) -> DriverState:
+        return self.config.get_driver_state(
+            quantity_factory=quantity_factory, communicator=communicator
+        )
+
+    @classmethod
+    def from_dict(cls, config: dict):
+        instance = cls.registry.from_dict(config)
+        return cls(config=instance, type=config["type"])
+
+
+@InitializationConfig.register("baroclinic")
+@dataclasses.dataclass
+class BaroclinicConfig(Initializer):
```

Comment:
Can you explain why we need both `Initializer` and `InitializationConfig`?

Reply:
Yes, the core reason is that there must be a class to represent each level of the yaml file. So the "Config" class represents the
```
type: my_type
config: <next level>
```
part of the yaml. The "next level" part is represented by Initializer.

We can maybe talk about what would be more informative names for these concepts, a difficulty is that they are both configuration. The MPI/comm configuration classes have the same structure/issue.